bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java

```diff
@@ -233,8 +232,16 @@ public void processResponse(@Nullable Response response) {
                     case COIOT_OPTION_GLOBAL_DEVID:
                         devId = opt.getStringValue();
                         String sVersion = substringAfterLast(devId, "#");
-                        int iVersion = Integer.parseInt(sVersion);
-                        if (coiotBound && (coiotVers != iVersion)) {
+                        int iVersion;
+                        try {
+                            iVersion = Integer.parseInt(sVersion);
+                        } catch (NumberFormatException e) {
+                            logger.debug("{}: Unable to parse version in CoIoT message: {}", thingName, devId);
+                            resetSerial();
```

Comment:
In don't see resetSerial() here, if a single packet can't be parsed this might doesn't impact the serial counter

Reply:
That's a mistake. I copied it from:

https://github.com/openhab/openhab-addons/blob/c417fe3f2f601e34a7b3a123b855054c9eeac585/bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api1/Shelly1CoapHandler.java#L307-L311

but actually that did not catch `NumberFormatException`. It looks like this would previously crash the binding. Should I just remove `resetSerial()` or what do you want to do in this case?

Reply:
yup, just remove resetSerial()

Reply:
Done.